sunpy/net/attr.py

```diff
@@ -346,18 +339,23 @@ def collides(self, other):
         return isinstance(other, self.__class__)
 
     def __repr__(self):
+        obj_placeholder = " object "
         attr_reg = AttrMeta._attr_registry[self.__class__]
-        new_repr = object.__repr__(self).split(" object ")
+        new_repr = object.__repr__(self).split(obj_placeholder)
         # If somehow the idx isn't in the attr reg, we still want it to print it
         # repr without error.
         try:
             idx = attr_reg.name_long.index(self.value)
-            new_repr.insert(1, f"({self.value}: {attr_reg.desc[idx]})")
+            obj_value_repr = f"({self.value}: {attr_reg.desc[idx]})"
         except ValueError:
-            new_repr.insert(1, f": {self.value}")
-        new_repr.insert(2, " object ")
+            obj_value_repr = f": {self.value}"
+        new_repr = [new_repr[0], obj_value_repr, obj_placeholder, new_repr[1]]
         return textwrap.fill("".join(new_repr), 100)
 
+    @property
+    def type_name(self):
+        return self.__class__.__name__.lower()
```

Comment:
This is to ensure Attrs work with the database ones?

Reply:
Yes, because `SimpleAttr`s are one of few `Attr`s that doesn't rely on their class name for the [creation](https://github.com/sunpy/sunpy/pull/4542/files#diff-78ef111a9bc1c3e7145f41fa4b2a5cf8R180-R183) from `ValueAttr`. This is because they are white-listed in the [following lines](https://github.com/sunpy/sunpy/blob/dc849c49a444ea17dacae3b397089ecb837e1511/sunpy/database/attrs.py#L15-L17).
I took advantage of this fact and inserted `type_name` (not a great name, I know) as a class attribute in the other `Attr`s, so to reduce string duplication and complexity, because properties share the same syntax.

Reply:
At least I already tested that this change doesn't break other walkers in the code :smiley: